### DISCUSSION TITLE
Should a lennoxicomfort.com Account Work with this integration?

### DISCUSSION THREAD

**Discussion by ElMoselYEE** [PeteRager/lennoxs30#329] [Q&A] [answered]
I just inherited an M30 and am trying to get it linked into HomeAssistant. When using the setup flow for this integration in HomeAssistant, my credentials that work on lennoxicomfort.com are rejected, and checking the logs I see that the integration is trying to access myicomfort.com. My credentials do not work on myicomfort.com, and I see no way to register a new myicomfort.com account, so I'm not sure what I'm doing wrong.

**Comment by PeteRager** [comment]
That's a good question.  The integration, as you stated, tries to login using this URL.

https://ic3messaging.myicomfort.com/v2/user/login

What error message is getting logged?  Does the mobile app work for you?

**Reply by ElMoselYEE** [reply, marked as answer]
Alright so I eventually got it to work by using a password that does not have special characters. 

I remember seeing some errors about not being able to have special characters when initially setting up the account, but it eventually accepted my password. My suspicion is myicomfort.com and lennoxicomfort.com use separate databases that sync between each other, and that sync has some rough edges.

For posterity the error I received in the logs was:

```
2024-06-23 10:39:01.135 ERROR (MainThread) [lennoxs30api.s30api_async] login failed response code [401] url [https://ic3messaging.myicomfort.com/v2/user/login] text [{"errors":[{"code":100200,"message":"Your username or password is incorrect.","doNotPersist":false}]}]
```

**Comment by PeteRager** [comment]
Glad it's working now.  From the readme

Passwords with special characters are not properly supported by the Lennox Cloud API and may not work. Known special characters that cause issues are & and ^. If your password contains these characters you may need to change it prior to using the integration
